sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/Utils.java

```diff
@@ -0,0 +1,228 @@
+
+        return new RepositoryProperties(
+            propsImpl.getName(),
+            propsImpl.getRegistryArtifactCount(),
+            propsImpl.getTagCount(),
+            writeableProps,
+            propsImpl.getCreatedOn(),
+            propsImpl.getLastUpdatedOn()
+        );
+    }
+
+    static RegistryArtifactProperties mapArtifactProperties(com.azure.containers.containerregistry.implementation.models.RegistryArtifactProperties propsImpl) {
+        if (propsImpl == null) {
+            return null;
+        }
+
+        List<RegistryArtifactProperties> registryArtifacts = getRegistryArtifacts(propsImpl.getRegistryArtifacts());
+
+        return new RegistryArtifactProperties(
+            propsImpl.getDigest(),
+            propsImpl.getManifestProperties(),
+            registryArtifacts,
+            propsImpl.getCpuArchitecture(),
+            propsImpl.getOperatingSystem(),
+            propsImpl.getCreatedOn(),
+            propsImpl.getLastUpdatedOn(),
+            propsImpl.getTags(),
+            propsImpl.getSize());
+    }
+
+    static ContentProperties mapContentProperties(com.azure.containers.containerregistry.implementation.models.ContentProperties props) {
+
+        if (props == null) {
+            return null;
+        }
+
+        return new ContentProperties()
+            .setCanDelete(props.isCanDelete())
+            .setCanList(props.isCanList())
+            .setCanRead(props.isCanRead())
+            .setCanWrite(props.isCanWrite());
+    }
+
+    static List<RegistryArtifactProperties> getRegistryArtifacts(List<ManifestAttributesManifestReferences> artifacts) {
+        if (artifacts == null) {
+            return null;
+        }
+
+        return artifacts.stream()
+            .map(artifact -> new RegistryArtifactProperties(
+                artifact.getDigest(),
+                null,
+                null,
+                artifact.getCpuArchitecture(),
+                artifact.getOperatingSystem(),
+                null,
+                null,
+                null,
+                null
+            )).collect(Collectors.toList());
+    }
+
+    static List<RegistryArtifactProperties> getRegistryArtifactsProperties(List<ManifestAttributesBase> baseArtifacts)
+    {
+        if (baseArtifacts == null) {
+            return null;
+        }
+
+        return baseArtifacts.stream().map(value -> new RegistryArtifactProperties(
+                value.getDigest(),
+                value.getWriteableProperties(),
+                getRegistryArtifacts(value.getReferences()),
+                value.getCpuArchitecture(),
+                value.getOperatingSystem(),
+                value.getCreatedOn(),
+                value.getLastUpdatedOn(),
+                value.getTags(),
+                value.getSize()
+            )
+
+        ).collect(Collectors.toList());
+    }
+
+    static TagProperties mapTagProperties(com.azure.containers.containerregistry.implementation.models.TagProperties props) {
```

Comment:
Mind documenting the difference between the generated and public classes? This will help in a few scenarios

- If generated classes change it'll be easier to determine how to update the hand-written public class
- It will note places where either code generation needs changing or a remind of locations where code generation customizations can be used to make the changes during generation time instead of at runtime